sycl/test/reduction/reduction_nd_s0_dw.cpp

```diff
@@ -1,7 +1,7 @@
 // UNSUPPORTED: cuda
 // Reductions use work-group builtins not yet supported by CUDA.
 //
-// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -fsycl-device-code-split=per_kernel -o %t.out
```

Comment:
Is the `-fsycl-device-code-split=per_kernel` required as it is not supported by CUDA (on top of the work-group built-in) and might make enabling the test in the future harder as both features are required?

Reply:
I discovered that 'has_extension("cl_khr_double")' helps to not run the code with 'double' type if device does not support 'double', but the test still fails without -fsycl-device-code-split=per_kernel.

Ok, the alternatives to this PR is a) not testing 'double' and 'half' at all, or b) move them to separate test(s).